Go 1.21 now includes the full version with `go mod init`

https://github.com/golang/go/issues/61851

--- a/main_test.go
+++ b/main_test.go
@@ -397,7 +397,6 @@
 
 		goVersion := runtime.Version()
-		// Strip all but the major and minor version.
-		goVersion = regexp.MustCompile(`^go(\d+\.\d+)`).FindStringSubmatch(goVersion)[1]
+		goVersion = regexp.MustCompile(`^go(\d+\.\d+\.\d+)`).FindStringSubmatch(goVersion)[1]
 		keyVals = append(keyVals, "GOVERSION", goVersion)
 		envhelpers.SetEnvVars(&env.Vars, keyVals...)
 
